projects/graph/src/draw.py

```diff
@@ -2,13 +2,57 @@
 General drawing methods for graphs using Bokeh.
 """
 
+import random
 from bokeh.io import show, output_file
 from bokeh.plotting import figure
 from bokeh.models import (GraphRenderer, StaticLayoutProvider, Circle, LabelSet,
-                          ColumnDataSource)
+                            ColumnDataSource)
+from bokeh.palettes import Spectral8
+from graph import Graph
 
 
 class BokehGraph:
     """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
+    def __init__(self, graph):
+        self.graph = graph
+
+    def show(self):
+        node_indices = list(self.graph.vertices.keys())
+        edges = [list(v) for v in self.graph.vertices.values()]
+
+        plot = figure(title='Graph Layout Demonstration', x_range=(-1, 5), y_range=(-1, 5),
+                      tools='', toolbar_location=None)
+        plot.axis.visible = False
+        plot.grid.visible = False
+
+        graph = GraphRenderer()
+
+        graph.node_renderer.data_source.add(node_indices, 'index')
+        graph.node_renderer.data_source.add(Spectral8, 'color')
+        graph.node_renderer.glyph = Circle(size=20, fill_color='color')
+
+        graph.edge_renderer.data_source.data = dict(
+            start=node_indices,
+            end=edges)
+
+        position = {}
+        for vertex in self.graph.vertices:
+            position[vertex] = (random.random() * 4, random.random() * 4)
+
```

Comment:
Great job! The colors and shapes and vertices are fun to play around with - trying out different things to see what happens can also help you learn about bokeh. The documentation is kind of a rabbit hole you can really get lost in.